src/tests/JIT/Regression/JitBlue/Runtime_44266/Runtime_44266.il

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This test shows an inlining of `byref LCL_VAR_ADDR - byref CNST_INT` method that returns a native int.
+// However, Jit could try to optimize `-` as `+ -CNST_INT` that could lead to an incorrect `long + (-byref)`.
+
+.assembly extern System.Console {}
+.assembly extern legacy library mscorlib {}
+.assembly byrefsubbyref1 { }
+.class a extends [mscorlib]System.Object
+{
+  .field static class ctest S_1
+  .method public static native int byrefsubbyref(class ctest& V_1, class ctest& V_2) aggressiveinlining
+  {
+    ldarg 0
+    ldarg 1
+    sub
+    ret
+  }
+
+  .method public static int32 main() cil managed
+  {
+    .entrypoint
+    .maxstack  2
+    .locals init (class ctest V_1,
+             class ctest V_2,
+             native int V_3)
+    newobj     instance void ctest::.ctor()
+    stloc.0
+    newobj     instance void ctest::.ctor()
+    dup
+    stsfld class ctest a::S_1
+    stloc.1
+
+    ldloca V_1
+    ldsflda class ctest a::S_1
+    call native int a::byrefsubbyref(class ctest&, class ctest&)
+    stloc V_3
+    ldloc V_3
+    ret
```

Comment:
Do you need conv.i4 before returning the value? Stack has `native int`, but method returns `int32`.

If you need the method to return 100, you could allocate a 100 element array and subtract the managed reference between the first element and last element.

The difference between a stack location and a static field that this is returning right now is a very random number.

Reply:
I am not an expert in ECMA so I am very glad that you are looking at the test, thank you.
My understanding is that ECMA allows implicit coercion of `native int` to `int` so I do not need a cast there.
However, I could read it wrong and can add a cast just for safety.

from ECMA, page 38:
I.8.7.3 General assignment compatibility:
![image](https://user-images.githubusercontent.com/22642771/98225001-068cf580-1f09-11eb-830f-051e25f6fcf0.png)

Reply:
I believe that @sandreenko is correct that this implicit conversion is allowed.